hk2-extras/pom.xml

```diff
@@ -91,6 +91,13 @@
                         <goals>
                             <goal>manifest</goal>
                         </goals>
+                        <configuration>
+                            <instructions>
+                                <Export-Package>
+                                    !*.internal,*;version=${project.osgi.version}
```

Comment:
Doesn't this export all transitive dependencies except `*.internal`?

Reply:
This is what it generates in the manifest:

```
Export-Package: org.glassfish.hk2.extras;version="3.1.1";uses:="org.glas
 sfish.hk2.api",org.glassfish.hk2.extras.events;version="3.1.1";uses:="o
 rg.glassfish.hk2.api,org.glassfish.hk2.api.messaging,org.jvnet.hk2.anno
 tations",org.glassfish.hk2.extras.interception;version="3.1.1";uses:="j
 akarta.inject,org.aopalliance.intercept,org.glassfish.hk2.api,org.jvnet
 .hk2.annotations",org.glassfish.hk2.extras.operation;version="3.1.1";us
 es:="org.glassfish.hk2.api,org.jvnet.hk2.annotations",org.glassfish.hk2
 .extras.provides;version="3.1.1";uses:="jakarta.inject,org.glassfish.hk
 2.api,org.jvnet.hk2.annotations"
Import-Package: jakarta.annotation;version="[3.0,4)",jakarta.inject;vers
 ion="[2.0,3)",org.aopalliance.intercept;version="[1.0,2)",org.glassfish
 .hk2.api;version="[3.1,4)",org.glassfish.hk2.api.messaging;version="[3.
 1,4)",org.glassfish.hk2.extras.events;version="[3.1,4)",org.glassfish.h
 k2.extras.interception;version="[3.1,4)",org.glassfish.hk2.utilities;ve
 rsion="[3.1,4)",org.glassfish.hk2.utilities.binding;version="[3.1,4)",o
 rg.glassfish.hk2.utilities.reflection;version="[3.1,4)",org.glassfish.h
 k2.utilities.reflection.internal;version="[3.1,4)",org.jvnet.hk2.annota
 tions;version="[3.1,4)"
```

I'm not an expert in OSGi, but I would say that transitive dependencies are not exported. They are only mentioned in the "uses" directive, but I don't think that means they are exported. how I understand it is that the "uses" directive only declares dependencies for exported packages.